src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -225,6 +225,41 @@ internal static void EmitJsonBodyOrServiceParameterPreparationString(this Endpoi
         codeWriter.EndBlock();
     }
 
+    internal static void EmitJsonBodyOrQueryParameterPreparationString(this EndpointParameter endpointParameter, CodeWriter codeWriter)
+    {
+        // Preamble for diagnostics purposes.
+        codeWriter.WriteLine(endpointParameter.EmitParameterDiagnosticComment());
+
+        // Declare handler variable up front.
+        codeWriter.WriteLine($"{endpointParameter.Type.ToDisplayString(EmitterConstants.DisplayFormat)} {endpointParameter.EmitHandlerArgument()} = null!;");
```

Comment:
Could we cache this resolver in the `RequestDelegateFactoryFunc` instead of doing the check with every request?

Reply:
Where is there to cache? Result from DisableInferBodyFromParameters property lookup?